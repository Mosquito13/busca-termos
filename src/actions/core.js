import actionTypes from '../constants/actionTypes';
import apiUtils from '../utils/apiUtils';

const coreActions = {
  setLoading(loading) {
    return {
      type: actionTypes.SET_LOADING,
      loading
    };
  },

  setData(data) {
    return {
      type: actionTypes.SET_DATA,
      data
    };
  },

  setSelectedItemId(id) {
    return {
      type: actionTypes.SET_SELECTED_ITEM_ID,
      id
    };
  },

  setFilter(filter) {
    return {
      type: actionTypes.SET_FILTER,
      filter
    };
  },

  loadData(settings) {
    return async (dispatch) => {
      const data = await apiUtils.loadData(settings?.languageFolder);

      dispatch(this.setData(data));
      dispatch(this.setLoading(false));
    };
  }
};

export default coreActions;
