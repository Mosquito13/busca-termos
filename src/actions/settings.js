import apiUtils from '../utils/apiUtils';
import actionTypes from '../constants/actionTypes';
import storageUtils from '../utils/storageUtils';
import coreActions from './core';

const settingsActions = {
  setValid(value) {
    return {
      type: actionTypes.SET_VALID,
      value
    };
  },

  setValidating(value) {
    return {
      type: actionTypes.SET_VALIDATING,
      value
    };
  },

  setLanguageFolder(value) {
    return {
      type: actionTypes.SET_LANGUAGE_FOLDER,
      value
    };
  },

  setLanguageFolderError(value) {
    return {
      type: actionTypes.SET_LANGUAGE_FOLDER_ERROR,
      value
    };
  },

  setMainLanguage(value) {
    return {
      type: actionTypes.SET_MAIN_LANGUAGE,
      value
    };
  },

  toggleTranslation(id, value) {
    return {
      type: actionTypes.TOGGLE_TRANSLATION,
      id,
      value
    };
  },

  toggleCompactLayout(value) {
    return {
      type: actionTypes.TOGGLE_COMPACT_LAYOUT,
      value
    };
  },

  toggleDarkTheme(value) {
    return {
      type: actionTypes.TOGGLE_DARK_THEME,
      value
    };
  },

  setSettings(value) {
    return {
      type: actionTypes.SET_SETTINGS,
      value
    };
  },

  validateAndSaveFirstSettings(languageFolder) {
    return async (dispatch) => {
      dispatch(this.setValidating(true));

      const isValid = await apiUtils.validateLanguageFolder(languageFolder);

      if (!isValid) {
        dispatch(this.setLanguageFolderError('Caminho invÃ¡lido.'));
        dispatch(this.setValid(false));
      } else {
        dispatch(coreActions.setLoading(true));
        dispatch(this.setValid(true));
        dispatch(this.saveSettings({ languageFolder }));
      }

      dispatch(this.setValidating(false));
    };
  },

  loadSettings() {
    return async (dispatch) => {
      const settings = storageUtils.getSettings();
      const isValid = await apiUtils.validateLanguageFolder(settings?.languageFolder);

      if (!isValid) {
        dispatch(this.setValid(false));
        dispatch(coreActions.setLoading(false));
      } else {
        dispatch(this.setSettings(settings));
        dispatch(coreActions.loadData(settings));
      }
    };
  },

  saveSettings(values) {
    return {
      type: actionTypes.SAVE,
      values
    };
  }
};

export default settingsActions;
