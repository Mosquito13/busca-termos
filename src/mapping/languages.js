import { ReactComponent as BrazilIcon } from '../assets/svg/brazil.svg';
import { ReactComponent as CataloniaIcon } from '../assets/svg/catalonia.svg';
import { ReactComponent as CzechIcon } from '../assets/svg/czech.svg';
import { ReactComponent as DenmarkIcon } from '../assets/svg/denmark.svg';
import { ReactComponent as FinlandIcon } from '../assets/svg/finland.svg';
import { ReactComponent as FranceIcon } from '../assets/svg/france.svg';
import { ReactComponent as GermanyIcon } from '../assets/svg/germany.svg';
import { ReactComponent as ItalyIcon } from '../assets/svg/italy.svg';
import { ReactComponent as PolandIcon } from '../assets/svg/poland.svg';
import { ReactComponent as PortugalIcon } from '../assets/svg/portugal.svg';
import { ReactComponent as RomaniaIcon } from '../assets/svg/romania.svg';
import { ReactComponent as RussiaIcon } from '../assets/svg/russia.svg';
import { ReactComponent as SlovakiaIcon } from '../assets/svg/slovakia.svg';
import { ReactComponent as SpainIcon } from '../assets/svg/spain.svg';
import { ReactComponent as TurkeyIcon } from '../assets/svg/turkey.svg';
import { ReactComponent as UsaIcon } from '../assets/svg/usa.svg';

const languageMapping = {
  BRAZIL: {
    id: 'pt-br',
    title: 'Português (Brasil)',
    getIcon: () => <BrazilIcon />
  },
  CATALONIA: {
    id: 'ca',
    title: 'Catalão',
    getIcon: () => <CataloniaIcon />
  },
  CZECH: {
    id: 'cs',
    title: 'Tcheco',
    getIcon: () => <CzechIcon />
  },
  DENMARK: {
    id: 'da',
    title: 'Dinamarquês',
    getIcon: () => <DenmarkIcon />
  },
  FINLAND: {
    id: 'fi',
    title: 'Finlandês',
    getIcon: () => <FinlandIcon />
  },
  FRANCE: {
    id: 'fr',
    title: 'Francês',
    getIcon: () => <FranceIcon />
  },
  GERMANY: {
    id: 'de',
    title: 'Alemão',
    getIcon: () => <GermanyIcon />
  },
  ITALY: {
    id: 'it',
    title: 'Italiano',
    getIcon: () => <ItalyIcon />
  },
  POLAND: {
    id: 'pl',
    title: 'Polonês',
    getIcon: () => <PolandIcon />
  },
  PORTUGAL: {
    id: 'pt',
    title: 'Português (Portugal)',
    getIcon: () => <PortugalIcon />
  },
  ROMANIA: {
    id: 'ro',
    title: 'Romeno',
    getIcon: () => <RomaniaIcon />
  },
  RUSSIA: {
    id: 'ru',
    title: 'Russo',
    getIcon: () => <RussiaIcon />
  },
  SLOVAKIA: {
    id: 'sk',
    title: 'Eslovaco',
    getIcon: () => <SlovakiaIcon />
  },
  SPAIN: {
    id: 'es',
    title: 'Espanhol',
    getIcon: () => <SpainIcon />
  },
  TURKEY: {
    id: 'tr',
    title: 'Turco',
    getIcon: () => <TurkeyIcon />
  },
  USA: {
    id: 'en-us',
    title: 'Inglês (EUA)',
    getIcon: () => <UsaIcon />
  }
};

export { languageMapping };
