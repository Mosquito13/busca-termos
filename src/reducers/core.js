import actionTypes from '../constants/actionTypes';

const initial = {
  loading: true,
  data: null,
  filter: '',
  selectedItemId: null,
  version: '',
  appHasUpdate: null,
  updateURL: ''
};

const coreReducer = (state = initial, action) => {
  switch (action.type) {
    case actionTypes.SET_LOADING:
      return { ...state, loading: action.loading };

    case actionTypes.SET_DATA:
      return { ...state, data: action.data };

    case actionTypes.SET_FILTER:
      return { ...state, filter: action.filter };

    case actionTypes.SET_APP_VERSION:
      return { ...state, version: action.version };

    case actionTypes.SET_APP_HAS_UPDATE:
      return { ...state, appHasUpdate: action.appHasUpdate, updateURL: action.updateURL };

    case actionTypes.SET_SELECTED_ITEM_ID:
      return { ...state, selectedItemId: action.id };

    default:
      return state;
  }
};

export default coreReducer;
