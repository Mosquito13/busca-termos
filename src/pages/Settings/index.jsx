import { useCallback } from 'react';
import { useDispatch, useSelector } from 'react-redux';
import { useNavigate } from 'react-router-dom';
import orderBy from 'lodash/orderBy';

import Icon, { SMALL, RESPONSIVE } from '../../components/common/Icon';
import Button from '../../components/common/Button';
import Toggle from '../../components/common/Toggle';
import Select from '../../components/common/Select';
import Fieldset from '../../components/common/Fieldset';
import { languageMapping } from '../../mapping/languages';

import coreActions from '../../actions/core';
import settingsActions from '../../actions/settings';
import settingsSelectors from '../../selectors/settings'

import './styles.scss';

const getLanguageItems = (mainLanguage, translation, onChangeTranslation) => {
  const languageToggles = [];
  const languageOptions = [];

  orderBy(languageMapping, ['title'])
    .forEach(({ id, title, getIcon }) => {
      languageToggles.push((
        <div key={id} className="settings__field">
          <Toggle
            label={title}
            icon={<Icon size={RESPONSIVE}>{getIcon()}</Icon>}
            value={translation[id] && mainLanguage !== id}
            onChange={() => onChangeTranslation(id, !translation[id])}
            disabled={mainLanguage === id}
          />
        </div>
      ));

      languageOptions.push({
        icon: <Icon size={SMALL}>{getIcon()}</Icon>,
        text: title,
        value: id
      });
    });

  return [languageToggles, languageOptions];
};

const Settings = () => {
  const navigate = useNavigate();
  const dispatch = useDispatch();
  const darkTheme = useSelector(settingsSelectors.isDarkTheme);
  const translation = useSelector(settingsSelectors.getTranslation);
  const mainLanguage = useSelector(settingsSelectors.getMainLanguage);

  const onClickApply = useCallback(() => {
    dispatch(coreActions.setLoading(true));
    dispatch(settingsActions.saveSettings({ mainLanguage, translation, darkTheme }));
    navigate(-1);
  }, [navigate, dispatch, mainLanguage, translation, darkTheme]);

  const onChangeDarkTheme = useCallback(() => {
    dispatch(settingsActions.toggleDarkTheme(!darkTheme));
  }, [dispatch, darkTheme]);

  const onChangeMainLanguage = useCallback((value) => {
    dispatch(settingsActions.setMainLanguage(value));
    dispatch(settingsActions.toggleTranslation(value, false));
  }, [dispatch]);

  const onChangeTranslation = useCallback((id, value) => {
    dispatch(settingsActions.toggleTranslation(id, value));
  }, [dispatch]);

  const [languageToggles, languageOptions] = getLanguageItems(mainLanguage, translation, onChangeTranslation);

  return (
    <div className="settings">
      <div className="settings__content">
        <Fieldset title="Aparência" noMarginTop>
          <div className="settings__field">
            <Toggle
              disabled
              label="Tema escuro"
              value={darkTheme}
              onChange={onChangeDarkTheme}
            />
          </div>
        </Fieldset>
        <Fieldset title="Idioma padrão">
          <div className="settings__field settings__field--50">
            <Select
              useRoundIconBorder
              options={languageOptions}
              value={mainLanguage}
              onChange={onChangeMainLanguage}
            />
          </div>
        </Fieldset>
        <Fieldset title="Mostrar no painel de traduções">
          <div className="settings__language-toggle-container">
            {languageToggles}
          </div>
        </Fieldset>
      </div>
      <div className="settings__footer">
        <Button
          text="Aplicar"
          onClick={onClickApply}
        />
      </div>
    </div>
  );
};

export default Settings;