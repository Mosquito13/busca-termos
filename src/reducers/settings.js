import actionTypes from '../constants/actionTypes';
import storageUtils from '../utils/storageUtils';
import { languageMapping } from '../mapping/languages';

const translationInitial = {};

Object.values(languageMapping).forEach(({ id }) => translationInitial[id] = true);

const initial = {
  /* internal controls */
  valid: true,
  validating: false,
  languageFolderError: '',
  /* saved settings */
  languageFolder: '',
  mainLanguage: languageMapping.BRAZIL.id,
  translation: translationInitial,
  compactLayout: false,
  darkTheme: true
};

const getNewSettings = (state, values) => {
  const { languageFolder, mainLanguage, translation, compactLayout, darkTheme } = state;
  const currentSettings = {
    languageFolder,
    compactLayout,
    mainLanguage,
    translation,
    darkTheme
  };

  return {
    ...currentSettings,
    ...values
  };
};

const settingsReducer = (state = initial, action) => {
  switch (action.type) {
    case actionTypes.SET_VALID:
      return { ...state, valid: action.value };

    case actionTypes.SET_VALIDATING:
      return { ...state, validating: action.value };

    case actionTypes.SET_MAIN_LANGUAGE:
      return { ...state, mainLanguage: action.value };

    case actionTypes.TOGGLE_COMPACT_LAYOUT:
      return { ...state, compactLayout: action.value };

    case actionTypes.TOGGLE_DARK_THEME:
      return { ...state, darkTheme: action.value };

    case actionTypes.TOGGLE_TRANSLATION:
      return {
        ...state,
        translation: {
          ...state.translation,
          [action.id]: action.value
        }
      };

    case actionTypes.SET_LANGUAGE_FOLDER:
      return { ...state, languageFolder: action.value };

    case actionTypes.SET_LANGUAGE_FOLDER_ERROR:
      return { ...state, languageFolderError: action.value };

    case actionTypes.SET_SETTINGS:
      return { ...state, ...getNewSettings(state, action.value) };

    case actionTypes.SAVE:
      storageUtils.setSettings(getNewSettings(state, action.values));
      return { ...state, languageFolderError: '' };

    default:
      return state;
  }
};

export default settingsReducer;
