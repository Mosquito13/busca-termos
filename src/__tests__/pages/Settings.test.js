import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';
import * as reactRedux from 'react-redux';

import Settings from '../../pages/Settings';
import * as coreActions from '../../actions/core';
import * as settingsActions from '../../actions/settings';
import { languageMapping } from '../../mapping/languages';

const navigateMock = jest.fn();
const reactRouterDomMock = { useNavigate: () => navigateMock };

jest.mock('react-router-dom', () => reactRouterDomMock);

const translationInitial = {};

Object.values(languageMapping).forEach(
  ({ id }) => (translationInitial[id] = true)
);

const getStateMock = () => ({
  Settings: {
    darkTheme: false,
    compactLayout: false,
    translationColumns: 3,
    translation: translationInitial,
    mainLanguage: languageMapping.USA.id
  }
});

const prepareMocks = (dispatchMock, stateMock) => {
  jest.spyOn(reactRedux, 'useDispatch').mockImplementation(() => dispatchMock);
  jest
    .spyOn(reactRedux, 'useSelector')
    .mockImplementation((selector) => selector(stateMock));
};

describe(`${Settings.name}`, () => {
  describe('tests', () => {
    let wrapper, dispatchMock;

    beforeAll(() => {
      dispatchMock = jest.fn();

      prepareMocks(dispatchMock, getStateMock());

      jest.spyOn(settingsActions, 'saveSettings');
      jest.spyOn(settingsActions, 'setMainLanguage');
      jest.spyOn(settingsActions, 'toggleCompactLayout');
      jest.spyOn(settingsActions, 'toggleDarkTheme');
      jest.spyOn(settingsActions, 'toggleTranslation');
      jest.spyOn(settingsActions, 'setTranslationColumns');
      jest.spyOn(coreActions, 'setLoading');

      wrapper = shallow(<Settings />);
    });

    it('should render', () => {
      expect(wrapper.find('.settings')).toHaveLength(1);
      expect(wrapper.find('[data-testid="toggle-dark-theme"]')).toHaveLength(1);
      expect(wrapper.find('[data-testid="toggle-compact-layout"]')).toHaveLength(1);
      expect(wrapper.find('[data-testid="select-main-lang"]')).toHaveLength(1);
      expect(wrapper.find('[data-testid="toggle-translation"]')).toHaveLength(16);
    });

    it('should toggle dark theme', () => {
      wrapper.find('[data-testid="toggle-dark-theme"]').simulate('change');
      expect(dispatchMock).toBeCalledTimes(1);
      expect(dispatchMock).toBeCalledWith(settingsActions.toggleDarkTheme(true));
    });

    it('should toggle compact layout', () => {
      wrapper.find('[data-testid="toggle-compact-layout"]').simulate('change');
      expect(dispatchMock).toBeCalledTimes(1);
      expect(dispatchMock).toBeCalledWith(settingsActions.toggleCompactLayout(true));
    });

    it('should change main language', () => {
      const czechId = languageMapping.CZECH.id;

      wrapper.find('[data-testid="select-main-lang"]').simulate('change', czechId);
      expect(dispatchMock).toBeCalledTimes(2);
      expect(dispatchMock).nthCalledWith(1, settingsActions.setMainLanguage(czechId));
      expect(dispatchMock).nthCalledWith(2, settingsActions.toggleTranslation(czechId, false));
    });

    it('should change translation', () => {
      const turkeyId = languageMapping.TURKEY.id;

      wrapper.find('[data-testid="toggle-translation"]').at(15).simulate('change');
      expect(dispatchMock).toBeCalledTimes(1);
      expect(dispatchMock).toBeCalledWith(settingsActions.toggleTranslation(turkeyId, false));
    });

    it('should change translation columns', () => {
      wrapper.find('[data-testid="translation-layout-selector"]').simulate('change', 2);
      expect(dispatchMock).toBeCalledTimes(1);
      expect(dispatchMock).toBeCalledWith(settingsActions.setTranslationColumns(2));
    });

    it('should save and navigate back', () => {
      wrapper.find('[data-testid="btn-back"]').simulate('click');
      expect(dispatchMock).toBeCalledTimes(2);
      expect(dispatchMock).nthCalledWith(1, coreActions.setLoading(true));
      expect(dispatchMock).nthCalledWith(2, settingsActions.saveSettings(
        languageMapping.USA.id,
        translationInitial,
        false,
        false
      ));
      expect(navigateMock).toBeCalledTimes(1);
      expect(navigateMock).toBeCalledWith(-1);
    });

    afterAll(() => {
      wrapper.unmount();
    });
  });

  describe('snapshots', () => {
    it('should render', () => {
      prepareMocks(jest.fn(), getStateMock());

      const tree = renderer.create(<Settings />).toJSON();

      expect(tree).toMatchSnapshot();
    });
  });
});
