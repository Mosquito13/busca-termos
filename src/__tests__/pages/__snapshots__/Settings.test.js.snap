// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Settings snapshots should render 1`] = `
<div
  className="settings"
>
  <div
    className="settings__content"
  >
    <fieldset
      className="fieldset fieldset--no-margin-top"
    >
      <legend
        className="fieldset__title"
      >
        Aparência
      </legend>
      <div
        className="settings__field"
      >
        <div
          className="toggle"
        >
          <div
            className="toggle__input"
            onClick={[Function]}
          >
            <div
              className="toggle__circle"
            >
              
            </div>
          </div>
          <div
            className="toggle__label"
            onClick={[Function]}
          >
            Tema escuro
          </div>
        </div>
      </div>
      <div
        className="settings__field"
      >
        <div
          className="toggle"
        >
          <div
            className="toggle__input"
            onClick={[Function]}
          >
            <div
              className="toggle__circle"
            >
              
            </div>
          </div>
          <div
            className="toggle__label"
            onClick={[Function]}
          >
            Tema compacto
          </div>
        </div>
      </div>
    </fieldset>
    <fieldset
      className="fieldset"
    >
      <legend
        className="fieldset__title"
      >
        Idioma padrão
      </legend>
      <div
        className="settings__field settings__field--50"
      >
        <div
          className="select"
        >
          <div
            className="select__field-container"
            onClick={[Function]}
          >
            <div
              className="select__field"
            >
              <div
                className="select__option select__option--unselectable"
              >
                <div
                  className="select__option-icon select__option-icon--rounded-border"
                >
                  <div
                    className="icon icon--small"
                  >
                    <svg>
                      usa.svg
                    </svg>
                  </div>
                </div>
                <div
                  className="select__option-text"
                >
                  Inglês (EUA)
                </div>
              </div>
            </div>
            <div
              className="select__field-icon"
            >
              <div
                className="icon icon--medium"
              >
                <svg
                  fill="none"
                  height="1em"
                  stroke="currentColor"
                  strokeLinecap="round"
                  strokeLinejoin="round"
                  strokeWidth="2"
                  style={
                    Object {
                      "color": undefined,
                    }
                  }
                  viewBox="0 0 24 24"
                  width="1em"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <polyline
                    points="6 9 12 15 18 9"
                  />
                </svg>
              </div>
            </div>
          </div>
          <div
            className="select__dropdown"
          >
            <div
              className="select__option"
              onClick={[Function]}
            >
              <div
                className="select__option-icon select__option-icon--rounded-border"
              >
                <div
                  className="icon icon--small"
                >
                  <svg>
                    germany.svg
                  </svg>
                </div>
              </div>
              <div
                className="select__option-text"
              >
                Alemão
              </div>
            </div>
            <div
              className="select__option"
              onClick={[Function]}
            >
              <div
                className="select__option-icon select__option-icon--rounded-border"
              >
                <div
                  className="icon icon--small"
                >
                  <svg>
                    catalonia.svg
                  </svg>
                </div>
              </div>
              <div
                className="select__option-text"
              >
                Catalão
              </div>
            </div>
            <div
              className="select__option"
              onClick={[Function]}
            >
              <div
                className="select__option-icon select__option-icon--rounded-border"
              >
                <div
                  className="icon icon--small"
                >
                  <svg>
                    denmark.svg
                  </svg>
                </div>
              </div>
              <div
                className="select__option-text"
              >
                Dinamarquês
              </div>
            </div>
            <div
              className="select__option"
              onClick={[Function]}
            >
              <div
                className="select__option-icon select__option-icon--rounded-border"
              >
                <div
                  className="icon icon--small"
                >
                  <svg>
                    slovakia.svg
                  </svg>
                </div>
              </div>
              <div
                className="select__option-text"
              >
                Eslovaco
              </div>
            </div>
            <div
              className="select__option"
              onClick={[Function]}
            >
              <div
                className="select__option-icon select__option-icon--rounded-border"
              >
                <div
                  className="icon icon--small"
                >
                  <svg>
                    spain.svg
                  </svg>
                </div>
              </div>
              <div
                className="select__option-text"
              >
                Espanhol
              </div>
            </div>
            <div
              className="select__option"
              onClick={[Function]}
            >
              <div
                className="select__option-icon select__option-icon--rounded-border"
              >
                <div
                  className="icon icon--small"
                >
                  <svg>
                    finland.svg
                  </svg>
                </div>
              </div>
              <div
                className="select__option-text"
              >
                Finlandês
              </div>
            </div>
            <div
              className="select__option"
              onClick={[Function]}
            >
              <div
                className="select__option-icon select__option-icon--rounded-border"
              >
                <div
                  className="icon icon--small"
                >
                  <svg>
                    france.svg
                  </svg>
                </div>
              </div>
              <div
                className="select__option-text"
              >
                Francês
              </div>
            </div>
            <div
              className="select__option select__option--selected"
              onClick={[Function]}
            >
              <div
                className="select__option-icon select__option-icon--rounded-border"
              >
                <div
                  className="icon icon--small"
                >
                  <svg>
                    usa.svg
                  </svg>
                </div>
              </div>
              <div
                className="select__option-text"
              >
                Inglês (EUA)
              </div>
            </div>
            <div
              className="select__option"
              onClick={[Function]}
            >
              <div
                className="select__option-icon select__option-icon--rounded-border"
              >
                <div
                  className="icon icon--small"
                >
                  <svg>
                    italy.svg
                  </svg>
                </div>
              </div>
              <div
                className="select__option-text"
              >
                Italiano
              </div>
            </div>
            <div
              className="select__option"
              onClick={[Function]}
            >
              <div
                className="select__option-icon select__option-icon--rounded-border"
              >
                <div
                  className="icon icon--small"
                >
                  <svg>
                    poland.svg
                  </svg>
                </div>
              </div>
              <div
                className="select__option-text"
              >
                Polonês
              </div>
            </div>
            <div
              className="select__option"
              onClick={[Function]}
            >
              <div
                className="select__option-icon select__option-icon--rounded-border"
              >
                <div
                  className="icon icon--small"
                >
                  <svg>
                    brazil.svg
                  </svg>
                </div>
              </div>
              <div
                className="select__option-text"
              >
                Português (Brasil)
              </div>
            </div>
            <div
              className="select__option"
              onClick={[Function]}
            >
              <div
                className="select__option-icon select__option-icon--rounded-border"
              >
                <div
                  className="icon icon--small"
                >
                  <svg>
                    portugal.svg
                  </svg>
                </div>
              </div>
              <div
                className="select__option-text"
              >
                Português (Portugal)
              </div>
            </div>
            <div
              className="select__option"
              onClick={[Function]}
            >
              <div
                className="select__option-icon select__option-icon--rounded-border"
              >
                <div
                  className="icon icon--small"
                >
                  <svg>
                    romania.svg
                  </svg>
                </div>
              </div>
              <div
                className="select__option-text"
              >
                Romeno
              </div>
            </div>
            <div
              className="select__option"
              onClick={[Function]}
            >
              <div
                className="select__option-icon select__option-icon--rounded-border"
              >
                <div
                  className="icon icon--small"
                >
                  <svg>
                    russia.svg
                  </svg>
                </div>
              </div>
              <div
                className="select__option-text"
              >
                Russo
              </div>
            </div>
            <div
              className="select__option"
              onClick={[Function]}
            >
              <div
                className="select__option-icon select__option-icon--rounded-border"
              >
                <div
                  className="icon icon--small"
                >
                  <svg>
                    czech.svg
                  </svg>
                </div>
              </div>
              <div
                className="select__option-text"
              >
                Tcheco
              </div>
            </div>
            <div
              className="select__option"
              onClick={[Function]}
            >
              <div
                className="select__option-icon select__option-icon--rounded-border"
              >
                <div
                  className="icon icon--small"
                >
                  <svg>
                    turkey.svg
                  </svg>
                </div>
              </div>
              <div
                className="select__option-text"
              >
                Turco
              </div>
            </div>
          </div>
        </div>
      </div>
    </fieldset>
    <fieldset
      className="fieldset"
    >
      <legend
        className="fieldset__title"
      >
        Mostrar no painel de traduções
      </legend>
      <div
        className="settings__language-toggle-container"
      >
        <div
          className="settings__field"
        >
          <div
            className="toggle"
          >
            <div
              className="toggle__input toggle__input--checked"
              onClick={[Function]}
            >
              <div
                className="toggle__circle toggle__circle--with-icon toggle__circle--checked"
              >
                <div
                  className="icon"
                >
                  <svg>
                    germany.svg
                  </svg>
                </div>
              </div>
            </div>
            <div
              className="toggle__label"
              onClick={[Function]}
            >
              Alemão
            </div>
          </div>
        </div>
        <div
          className="settings__field"
        >
          <div
            className="toggle"
          >
            <div
              className="toggle__input toggle__input--checked"
              onClick={[Function]}
            >
              <div
                className="toggle__circle toggle__circle--with-icon toggle__circle--checked"
              >
                <div
                  className="icon"
                >
                  <svg>
                    catalonia.svg
                  </svg>
                </div>
              </div>
            </div>
            <div
              className="toggle__label"
              onClick={[Function]}
            >
              Catalão
            </div>
          </div>
        </div>
        <div
          className="settings__field"
        >
          <div
            className="toggle"
          >
            <div
              className="toggle__input toggle__input--checked"
              onClick={[Function]}
            >
              <div
                className="toggle__circle toggle__circle--with-icon toggle__circle--checked"
              >
                <div
                  className="icon"
                >
                  <svg>
                    denmark.svg
                  </svg>
                </div>
              </div>
            </div>
            <div
              className="toggle__label"
              onClick={[Function]}
            >
              Dinamarquês
            </div>
          </div>
        </div>
        <div
          className="settings__field"
        >
          <div
            className="toggle"
          >
            <div
              className="toggle__input toggle__input--checked"
              onClick={[Function]}
            >
              <div
                className="toggle__circle toggle__circle--with-icon toggle__circle--checked"
              >
                <div
                  className="icon"
                >
                  <svg>
                    slovakia.svg
                  </svg>
                </div>
              </div>
            </div>
            <div
              className="toggle__label"
              onClick={[Function]}
            >
              Eslovaco
            </div>
          </div>
        </div>
        <div
          className="settings__field"
        >
          <div
            className="toggle"
          >
            <div
              className="toggle__input toggle__input--checked"
              onClick={[Function]}
            >
              <div
                className="toggle__circle toggle__circle--with-icon toggle__circle--checked"
              >
                <div
                  className="icon"
                >
                  <svg>
                    spain.svg
                  </svg>
                </div>
              </div>
            </div>
            <div
              className="toggle__label"
              onClick={[Function]}
            >
              Espanhol
            </div>
          </div>
        </div>
        <div
          className="settings__field"
        >
          <div
            className="toggle"
          >
            <div
              className="toggle__input toggle__input--checked"
              onClick={[Function]}
            >
              <div
                className="toggle__circle toggle__circle--with-icon toggle__circle--checked"
              >
                <div
                  className="icon"
                >
                  <svg>
                    finland.svg
                  </svg>
                </div>
              </div>
            </div>
            <div
              className="toggle__label"
              onClick={[Function]}
            >
              Finlandês
            </div>
          </div>
        </div>
        <div
          className="settings__field"
        >
          <div
            className="toggle"
          >
            <div
              className="toggle__input toggle__input--checked"
              onClick={[Function]}
            >
              <div
                className="toggle__circle toggle__circle--with-icon toggle__circle--checked"
              >
                <div
                  className="icon"
                >
                  <svg>
                    france.svg
                  </svg>
                </div>
              </div>
            </div>
            <div
              className="toggle__label"
              onClick={[Function]}
            >
              Francês
            </div>
          </div>
        </div>
        <div
          className="settings__field"
        >
          <div
            className="toggle"
          >
            <div
              className="toggle__input toggle__input--disabled"
              onClick={null}
            >
              <div
                className="toggle__circle toggle__circle--with-icon"
              >
                <div
                  className="icon"
                >
                  <svg>
                    usa.svg
                  </svg>
                </div>
              </div>
            </div>
            <div
              className="toggle__label toggle__label--disabled"
              onClick={null}
            >
              Inglês (EUA)
            </div>
          </div>
        </div>
        <div
          className="settings__field"
        >
          <div
            className="toggle"
          >
            <div
              className="toggle__input toggle__input--checked"
              onClick={[Function]}
            >
              <div
                className="toggle__circle toggle__circle--with-icon toggle__circle--checked"
              >
                <div
                  className="icon"
                >
                  <svg>
                    italy.svg
                  </svg>
                </div>
              </div>
            </div>
            <div
              className="toggle__label"
              onClick={[Function]}
            >
              Italiano
            </div>
          </div>
        </div>
        <div
          className="settings__field"
        >
          <div
            className="toggle"
          >
            <div
              className="toggle__input toggle__input--checked"
              onClick={[Function]}
            >
              <div
                className="toggle__circle toggle__circle--with-icon toggle__circle--checked"
              >
                <div
                  className="icon"
                >
                  <svg>
                    poland.svg
                  </svg>
                </div>
              </div>
            </div>
            <div
              className="toggle__label"
              onClick={[Function]}
            >
              Polonês
            </div>
          </div>
        </div>
        <div
          className="settings__field"
        >
          <div
            className="toggle"
          >
            <div
              className="toggle__input toggle__input--checked"
              onClick={[Function]}
            >
              <div
                className="toggle__circle toggle__circle--with-icon toggle__circle--checked"
              >
                <div
                  className="icon"
                >
                  <svg>
                    brazil.svg
                  </svg>
                </div>
              </div>
            </div>
            <div
              className="toggle__label"
              onClick={[Function]}
            >
              Português (Brasil)
            </div>
          </div>
        </div>
        <div
          className="settings__field"
        >
          <div
            className="toggle"
          >
            <div
              className="toggle__input toggle__input--checked"
              onClick={[Function]}
            >
              <div
                className="toggle__circle toggle__circle--with-icon toggle__circle--checked"
              >
                <div
                  className="icon"
                >
                  <svg>
                    portugal.svg
                  </svg>
                </div>
              </div>
            </div>
            <div
              className="toggle__label"
              onClick={[Function]}
            >
              Português (Portugal)
            </div>
          </div>
        </div>
        <div
          className="settings__field"
        >
          <div
            className="toggle"
          >
            <div
              className="toggle__input toggle__input--checked"
              onClick={[Function]}
            >
              <div
                className="toggle__circle toggle__circle--with-icon toggle__circle--checked"
              >
                <div
                  className="icon"
                >
                  <svg>
                    romania.svg
                  </svg>
                </div>
              </div>
            </div>
            <div
              className="toggle__label"
              onClick={[Function]}
            >
              Romeno
            </div>
          </div>
        </div>
        <div
          className="settings__field"
        >
          <div
            className="toggle"
          >
            <div
              className="toggle__input toggle__input--checked"
              onClick={[Function]}
            >
              <div
                className="toggle__circle toggle__circle--with-icon toggle__circle--checked"
              >
                <div
                  className="icon"
                >
                  <svg>
                    russia.svg
                  </svg>
                </div>
              </div>
            </div>
            <div
              className="toggle__label"
              onClick={[Function]}
            >
              Russo
            </div>
          </div>
        </div>
        <div
          className="settings__field"
        >
          <div
            className="toggle"
          >
            <div
              className="toggle__input toggle__input--checked"
              onClick={[Function]}
            >
              <div
                className="toggle__circle toggle__circle--with-icon toggle__circle--checked"
              >
                <div
                  className="icon"
                >
                  <svg>
                    czech.svg
                  </svg>
                </div>
              </div>
            </div>
            <div
              className="toggle__label"
              onClick={[Function]}
            >
              Tcheco
            </div>
          </div>
        </div>
        <div
          className="settings__field"
        >
          <div
            className="toggle"
          >
            <div
              className="toggle__input toggle__input--checked"
              onClick={[Function]}
            >
              <div
                className="toggle__circle toggle__circle--with-icon toggle__circle--checked"
              >
                <div
                  className="icon"
                >
                  <svg>
                    turkey.svg
                  </svg>
                </div>
              </div>
            </div>
            <div
              className="toggle__label"
              onClick={[Function]}
            >
              Turco
            </div>
          </div>
        </div>
      </div>
    </fieldset>
  </div>
  <div
    className="settings__footer"
  >
    <button
      className="button"
      onClick={[Function]}
    >
      <div
        className="button__content"
      >
        Voltar
      </div>
    </button>
  </div>
</div>
`;
